### PR TITLE
Upgrade Kyma to version 2.0

### PR DESCRIPTION
**Pull request by StefanCholakov**
**Description**

This PR bumps and adapts the kyma from 1.24.8 to 2.0

Changes proposed in this pull request:
- adapt usage of k3d instead of minikube on local installation
- delete obsolete 1.24.8 related installer resources
- adapt overrides from `.dot` style to helm native yaml tree format
- install compass with helm
- various adaptations across charts & installation scripts

**Pull Request status**

- [x] Implementation
- [ ] Unit tests
- [ ] Integration tests
- [x] `chart/compass/values.yaml` is updated 
- [ ] Mocks are regenerated, using the automated script

### CODE REVIEW

File: installation/resources/compass-overrides-local.yaml [hunk 1]
```diff
@@ -0,0 +1,205 @@
+config:
+  oidc:
+    caFile: "/etc/dex-tls-cert/tls.crt"
```
Comment (DimitarPetrov) [inline installation/resources/compass-overrides-local.yaml:3]: Probably a leftover dex reference?

File: installation/scripts/install-compass.sh [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+            shift # past argument
+        ;;
+    esac
+done
+set -- "${POSITIONAL[@]}" # restore positional parameters
+
+CRDS_FOLDER="${CURRENT_DIR}/../resources/crds"
+kubectl apply -f "${CRDS_FOLDER}"
+helm upgrade --install --wait --timeout "${TIMEOUT}" -f ./mergedOverrides.yaml --create-namespace --namespace compass-system compass "${COMPASS_CHARTS}"
+rm mergedOverrides.yaml
```
Comment (DimitarPetrov) [inline installation/scripts/install-compass.sh:49]: If something fail the file will be left behind.

Add it to gitignore for such cases.

Also you can use trap to minimize such cases: https://github.com/kyma-incubator/compass/blob/e495587b9928d436701650783fb9e453d03b424f/installation/cmd/run.sh#L184

**Review by NickyMateev — APPROVED** [review]
Looks good to me - at least I couldn't find anything disturbing 👍 

One question - how do things look for Mac M1 users - does the local k3d cluster compass installation work?

Also I noticed that `--k3d-cpus` was removed - I don't remember the original reason for adding it - maybe because there was an equivalent minikube-like option or that we were trying to stabilize things on M1 machines, by allowing the tweaking of the cpu. Anyway, the parameter is now removed - what were the reasons and thoughts behind that?

**Comment by nyordanoff** [comment]
> Looks good to me - at least I couldn't find anything disturbing 👍
> 
> One question - how do things look for Mac M1 users - does the local k3d cluster compass installation work?
> 
> Also I noticed that `--k3d-cpus` was removed - I don't remember the original reason for adding it - maybe because there was an equivalent minikube-like option or that we were trying to stabilize things on M1 machines, by allowing the tweaking of the cpu. Anyway, the parameter is now removed - what were the reasons and thoughts behind that?

М1 testing and adaptations were left out of scope for the current pass. We will adapt things ad-hoc if something comes up. 

Regarding the `--k3d-cpus` flag: currently, while provisioning minikube one can explicitly specify the number of cpus via flag. However, k3d does not support such flag (it supports only for memory) so the `cpu` flag to the script became obsolete. That's why it was removed.

I searched through k3d docs & code to check if somehow we can pass number of cpus during cluster creation but it seems that it is not supported. If I find a way to do it, I will add back the flag.

**Review by nkanchev — DISMISSED** [review]

**Review by NickyMateev — DISMISSED** [review]

**Review by desislavaa — APPROVED** [review]

**Review by StanislavStefanov — APPROVED** [review]

**Review by desislavaa — APPROVED** [review]

**Review by StanislavStefanov — APPROVED** [review]

**Review by NickyMateev — APPROVED** [review]

**Review by nkanchev — APPROVED** [review]

**Pull request merged**